### ISSUE TITLE
List concatenation not resolving correctly to `list[str]` but to `list[_T]`

### ISSUE THREAD

**Issue by ion-elgreco** [astral-sh/ty#314]
### Summary

```python
def print_list(value: list[str]) -> None:
    print(value)


existing_list = ["foo"]
print_list(
    existing_list
    + [
        "bar",
    ],
)
```

```python
  --> test.py:7:5
   |
 5 |   existing_list = ["foo"]
 6 |   print_list(
 7 | /     existing_list
 8 | |     + [
 9 | |         "bar",
10 | |     ],
   | |_____^ Expected `list[str]`, found `list[_T]`
11 |   )
   |
info: Function defined here
 --> test.py:1:5
  |
1 | def print_list(value: list[str]) -> None:
  |     ^^^^^^^^^^ ---------------- Parameter declared here
2 |     print(value)
  |
info: `invalid-argument-type` is enabled by default

Found 1 diagnostic
```

**Comment by hauntsaninja** [comment]
I think ty just doesn't support any list concatenation right now:
```
error[unsupported-operator]: Operator `+` is unsupported between objects of type `list[int]` and `list[int]`
  --> test1.py:14:12
   |
13 | def f4(x: list[int], y: list[int]):
14 |     return x + y
   |            ^^^^^
   |
info: `unsupported-operator` is enabled by default
```
